File: app/src/main/java/com/vikingsen/cheesedemo/model/webservice/CheeseService.java
package com.vikingsen.cheesedemo.model.webservice;


import com.vikingsen.cheesedemo.model.webservice.dto.CheeseDto;
import com.vikingsen.cheesedemo.model.webservice.dto.CommentDto;
import com.vikingsen.cheesedemo.model.webservice.dto.CommentRequestDto;
import com.vikingsen.cheesedemo.model.webservice.dto.CommentResponse;
import com.vikingsen.cheesedemo.model.webservice.dto.PriceDto;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface CheeseService {

    @GET("api/v1/cheese")
    Call<List<CheeseDto>> getCheeses();

    @GET("api/v1/cheese/{cheeseId}")
    Call<CheeseDto> getCheese(@Path("cheeseId") long cheeseId);

    @GET("api/v1/price/{cheeseId}")
    Call<PriceDto> getPrice(@Path("cheeseId") long cheeseId);

    @GET("api/v1/comment/{cheeseId}")
    Call<List<CommentDto>> getComments(@Path("cheeseId") long cheeseId);

    @POST("api/v1/comment")
    Call<List<CommentResponse>> postComment(@Body List<CommentRequestDto> commentRequest);
}

File: app/src/main/java/com/vikingsen/cheesedemo/model/webservice/CheeseService.kt
package com.vikingsen.cheesedemo.model.webservice


import com.vikingsen.cheesedemo.model.webservice.dto.CheeseDto
import com.vikingsen.cheesedemo.model.webservice.dto.CommentDto
import com.vikingsen.cheesedemo.model.webservice.dto.CommentRequestDto
import com.vikingsen.cheesedemo.model.webservice.dto.CommentResponse
import com.vikingsen.cheesedemo.model.webservice.dto.PriceDto

import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path

interface CheeseService {

    @GET("api/v1/cheese")
    fun getCheeses(): Call<List<CheeseDto>>

    @GET("api/v1/cheese/{cheeseId}")
    fun getCheese(@Path("cheeseId") cheeseId: Long): Call<CheeseDto>

    @GET("api/v1/price/{cheeseId}")
    fun getPrice(@Path("cheeseId") cheeseId: Long): Call<PriceDto>

    @GET("api/v1/comment/{cheeseId}")
    fun getComments(@Path("cheeseId") cheeseId: Long): Call<List<CommentDto>>

    @POST("api/v1/comment")
    fun postComment(@Body commentRequest: List<CommentRequestDto>): Call<List<CommentResponse>>
}
